### ISSUE TITLE
ENH: support a fixed bandwidth in multivariate gaussian_kde

### ISSUE THREAD

**Issue by martinfleis** [scipy/scipy#19250]
### Is your feature request related to a problem? Please describe.

At the moment, multivariate `gaussian_kde` selects a bandwidth automatically for each dimension separately. While this makes sense for a lot of use cases, it cannot be applied to coordinates representing points on a plane. This is different than what sklearn is doing in neighbors.KernelDensity, where a single bandwidth is used across all dimensions.

### Describe the solution you'd like.

I would like to have an option to specify not only the factor of the bandwidth but fix the bandwidth across all dimensions to a set scalar (i.e. meters in case of geographic points with x, y coordinates used as variables). As far as I can tell, there is no way achieving that right now.

### Describe alternatives you've considered.

I can use sklearn directly, which is often fine. However, the issue arised in seaborn (https://github.com/mwaskom/seaborn/issues/3472) which wraps `gaussian_kde` in its `kdeplot` and where adding sklearn as a dependency is a no-go.

### Additional context (e.g. screenshots, GIFs)

See the original issue in seaborn for visual representation of an issue and a comparison with a sklearn implementation -> https://github.com/mwaskom/seaborn/issues/3472

**Comment by mdhaber** [comment]
Does specifying a constant for `bw_factor` not do what you're looking for?

**Comment by martinfleis** [comment]
No, that seems to only scale the bandwidth per axis based on its extent, it does not fix the bandwidth.

**Comment by mdhaber** [comment]
Ok. This is a fair request, and it this looks pretty easy to do. Internally, `gaussian_kde`'s `_compute_covariance` method currently calculates the raw (not scaled by bandwidth factor) covariance matrix from the data. It sounds like you just want it to use the identity matrix. I'd just add `covariance` as a keyword parameter, if that sounds good.

**Comment by martinfleis** [comment]
> It sounds like you just want it to use the identity matrix. I'd just add `covariance` as a keyword parameter, if that sounds good.

Can't tell, that is too deep in the internals for me. `covariance` would take which options?

**Comment by mdhaber** [comment]
The `covariance` parameter would accept a covariance matrix (a symmetric positive definite array) as the argument. The identity matrix (`np.eye(d)`, where `d=2` in your 2d case) would work for your use case. `seaborn` would then update their UI to expose something related to the covariance as an option, but I'm not sure exactly what they'd want to do.

**Comment by martinfleis** [comment]
Ah, cool! Yes, that would work.

**Comment by RadostW** [comment]
I encountered another use case which requires exactly this functionality - I wanted to show ratio of two densities with two different weights. This requires having the same kernel for both smoothings. Currently this behavior can be emulated by using undocumented `cho_cov` property.

```python
kde = gaussian_kde(df[["x", "y"]].T, weights=df["z"].values)
kde_const = gaussian_kde(df[["x", "y"]].T)
kde_const.cho_cov = kde.cho_cov  # override covariance of const smoothing

# KDE on full grid
z_raw = kde(grid_points.T).reshape(nx, ny)
z_const = kde_const(grid_points.T).reshape(nx, ny)
z = z_raw / z_const
```

Having an option to pass a fixed, pre-computed covariance to `gaussian_kde.__init__` would solve the problem in a clean way.
